logic.js

```diff
@@ -0,0 +1,27 @@
+class Logic{
+
+    checkMatch(card1, card2){
+        if($(card1).data().id === $(card2).data().id && $(card1).data().gridArea !== $(card2).data().gridArea){
+            return true
+        } else{
+            return false
+        }
+    }
+
+    checkEndGame(){
+        if($('img').length ===0){
+            if(localStorage.bestScore === undefined){    ////For the first game user plays => just a big num which will be win for sure
```

Comment:
1. If `localStorage.bestScore` doesn't exist it will be undefined which is falsey. Therefore you can just check `if(!localStorage.bestScore)` basically is localstorage falsey.

2. You shouldn't make assumptions with setting a high number that the user will for sure win. You should just set localstorage to the current attempts and use an `else` statement for the other scenario.